From 73b1f292ed720beb00fd55b3365934ac74d2a49b Mon Sep 17 00:00:00 2001
From: andrew morton <drewish@katherinehouse.com>
Date: Mon, 9 Apr 2012 10:26:04 -0400
Subject: [PATCH] Issue #1213956 by lyricnz: Prevent automatic downloading of
 files.

---
 mediaelement.install |    2 ++
 mediaelement.module  |   15 +++++++++++++++
 2 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/mediaelement.install b/mediaelement.install
index 68cee3d..b596cb4 100644
--- a/mediaelement.install
+++ b/mediaelement.install
@@ -16,10 +16,12 @@ function mediaelement_uninstall() {
   variable_del('mediaelement_audio_default_class');
   variable_del('mediaelement_audio_default_download_link');
   variable_del('mediaelement_audio_default_download_text');
+  variable_del('mediaelement_audio_default_preload');
   variable_del('mediaelement_video_default_controls');
   variable_del('mediaelement_video_default_width');
   variable_del('mediaelement_video_default_height');
   variable_del('mediaelement_video_default_class');
   variable_del('mediaelement_video_default_download_link');
   variable_del('mediaelement_video_default_download_text');
+  variable_del('mediaelement_video_default_preload');
 }
diff --git a/mediaelement.module b/mediaelement.module
index 7be066f..9d87482 100644
--- a/mediaelement.module
+++ b/mediaelement.module
@@ -75,6 +75,7 @@ function mediaelement_field_formatter_info() {
       'height' => variable_get('mediaelement_video_default_height', '385'),
       'download_link' => variable_get('mediaelement_video_default_download_link', FALSE),
       'download_text' => variable_get('mediaelement_video_default_download_text', t('Download')),
+      'preload' => variable_get('mediaelement_video_default_preload', 'auto'),
     ),
   );
 
@@ -87,6 +88,7 @@ function mediaelement_field_formatter_info() {
       'height' => variable_get('mediaelement_audio_default_height', '30'),
       'download_link' => variable_get('mediaelement_audio_default_download_link', FALSE),
       'download_text' => variable_get('mediaelement_audio_default_download_text', t('Download')),
+      'preload' => variable_get('mediaelement_audio_default_preload', 'auto'),
     ),
   );
 
@@ -124,6 +126,7 @@ function mediaelement_field_formatter_view($entity_type, $entity, $field, $insta
       '#attributes' => array(
         'src' => file_create_url($src),
         'class' => $class,
+        'preload' => $display['settings']['preload'],
       ),
       '#settings' => $settings,
       '#attached' => array(
@@ -222,6 +225,18 @@ function mediaelement_field_formatter_settings_form($field, $instance, $view_mod
     '#default_value' => $settings['download_text'],
   );
 
+  $element['preload'] = array(
+    '#title' => t('Preload'),
+    '#description' => t('Provide a hint to the user agent about whether it should pre-load the media resource.'),
+    '#type' => 'radios',
+    '#options' => array(
+      'none' => t('None: Do not preload the resource.'),
+      'metadata' => t('Metadata: Load only the resource metadata (dimensions, first frame, track list, duration, etc). May cause full preload.'),
+      'auto' => t('Automatic: Preload is possible.'),
+    ),
+    '#default_value' => $settings['preload'],
+  );
+
   return $element;
 }
 
-- 
1.7.7.4

